falcon/asgi/request.py

```diff
@@ -837,14 +861,19 @@ def get_header(self, name, required=False, default=None):
 
         """
 
-        asgi_name = name.lower()
+        try:
+            asgi_name = _name_cache[name]
+        except KeyError:
+            asgi_name = name.lower().encode()
+            if len(_name_cache) < 64:  # Somewhat arbitrary ceiling to mitigate abuse
```

Comment:
this would keep only the first 64 entry forever. not sure if we could use popitem to create an lru-like cache?

I'm not sure if this would make the performance consideration above moot though

Reply:
My reasoning was just that a well-behaved app should actually request a much smaller set of request header names over its lifetime than 64, but just to prevent it from growing unbounded to put a sanity check in there.

Reply:
Yes I agree on that observation. Was just an observation. As mentioned I'm not sure this comment and the other one are worth acting on

Reply:
I've just verified the below still holds on Py3.8:
```python
>>> previous = 0
>>> for number in range(256):
...     d = {i: i*i for i in range(number)}
...     if sys.getsizeof(d) != previous:
...         previous = sys.getsizeof(d)
...         print(number, previous)
... 
0 64
1 232
6 360
11 640
22 1176
43 2272
86 4696
171 9312
```

Basically, the size of a Python `dict` hash table grows at these numbers of keys in the dict. So it would make sense to cap at 42, 85 etc.

Reply:
Cool, definitely something to keep in mind.

Reply:
My vote would go to 42 😂